### PR TITLE
First Comparison Script Telomerase Activity

### PR DESCRIPTION
**Pull request by NNoureen**
### Purpose/implementation Section
The analysis is focused on identifying telomerase activities using gene expression data sets processed in different ways, to identify platform differences and compare the telomerase activities across different tumor histologies.

#### What was your approach?
We used our newly developed method EXTEND (Expression based Telomerase ENzymatic activity Detection).

#### What GitHub issue does your pull request address?
Issue # 148 (Quantify telomerase activity across pediatric brain tumors)

#### Which areas should receive a particularly close look?
We would like the reviewers to provide us directions for further comparisons regarding telomerase activities if required.
#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
YES
#### What types of results are included (e.g., table, figure)?
Predicted Telomerase activity scores as tables and comparisons as figures.
#### What is your summary of the results?
Results include text files containing telomerase activities across different expression data formats. The comparison of telomerase activities has been carried out with different parameters as:
1.	Variations in telomerase activities across different expression data formats.
2.	Comparison of telomerase activities with TERT and TERC genes expression for different data sets.
3.	Distribution of telomerase activities across different pediatric brain tumor histologies.

#### Documentation Checklist

This analysis module has a `README` and it is up to date.
This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
The analytical code is documented.

### CODE REVIEW

**Comment by jaclyn-taroni** [comment]
Hi @NNoureen - I've asked @sjspielman, an OpenPBTA contributor, to provide a review. Thanks!

**Comment by NNoureen** [comment]
Sure. Thanks

File: analyses/telomerase-activity-prediction/02-Comparing-Counts-versus-FPKM.R [hunk 1]
```diff
@@ -0,0 +1,77 @@
+library('pracma')
```
Comment (sjspielman) [inline analyses/telomerase-activity-prediction/02-Comparing-Counts-versus-FPKM.R]: I think most of these packages are not needed - I successfully ran this script loading _only_ two: packages `ggpubr` and `grid`. Try and see how many of these library statements can be removed to get the minimal set of packages needed.

File: analyses/telomerase-activity-prediction/02-Comparing-Counts-versus-FPKM.R [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+colnames(PTBA_GE_PolyA_TMScores2)[colnames(PTBA_GE_PolyA_TMScores2)=="NormEXTENDScores"]='NormEXTENDScores_PolyA_FPKM'
+
+PTBA_GE_Standard_TMScores2 = read.table(D,sep='\t',head=T)
+colnames(PTBA_GE_Standard_TMScores2)[colnames(PTBA_GE_Standard_TMScores2)=="NormEXTENDScores"]='NormEXTENDScores_Stranded_FPKM'
+
+PTBA_GE_PolyA_TMScores = merge(PTBA_GE_PolyA_TMScores1,PTBA_GE_PolyA_TMScores2,by='SampleID')
+PTBA_GE_Standard_TMScores = merge(PTBA_GE_Standard_TMScores1,PTBA_GE_Standard_TMScores2,by='SampleID')
+
+
+pdf('plots/PTBA_GE_Score_AllScatter.pdf')
```
Comment (sjspielman) [inline analyses/telomerase-activity-prediction/02-Comparing-Counts-versus-FPKM.R]: I would recommend creating a variable at the top of the script with this file name, OR (and perhaps a better solution?), pass in the output plot name as a command line argument into the script, which you can integrate into the shell script.

File: analyses/telomerase-activity-prediction/02-Comparing-Counts-versus-FPKM.R [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+		   
+P2 = ggscatter(PTBA_GE_Standard_TMScores, x = "NormEXTENDScores_StrandedCounts", y = "NormEXTENDScores_Stranded_FPKM",color = "red",size = 1.3,
+add = "reg.line",  # Add regressin line
+add.params = list(color = "black", fill = "grey",size=0.5), # Customize reg. line
+conf.int = TRUE # Add confidence interval
+)+ stat_cor(method = "spearman",size=2)
+		   
+grid.newpage()
+# Create layout : nrow = 2, ncol =2
+pushViewport(viewport(layout = grid.layout(nrow = 2, ncol = 2)))
```
Comment (sjspielman) [inline analyses/telomerase-activity-prediction/02-Comparing-Counts-versus-FPKM.R]: Your output is only 2 plots, but the pane is for 2x2, leaving a lot of empty space in the resulting PDF. To fix this, you should make `nrow=1`, and you may (will?) need to fiddle with the PDF dimensions in your call to `pdf()` on line 47 by adding width and/or height arguments.

**Comment by sjspielman** [comment]
Hi @NNoureen! Overall this looks pretty good and just needs a few updates! I've left you some comments in the "Files Changed" section of the PR. In addition to these, we are trying to have only _one_ shell script for analysis, which in your case will be the script `RUN-telomerase-activity-prediction.sh`. This script has been integrated into the CI config file, but the new one has not, meaning the new analysis won't run in CI. Fixing this is quick! You just need to tack on all code from the new script `Comparing-Counts-versus-FPKM.sh` into the first shell script, and then remove the new script entirely. As we continue to build up analyses, you can add the new lines calling R scripts into that same original shell script.

**Comment by NNoureen** [comment]
Thanks. We will do the modifications.

**Comment by jaclyn-taroni** [comment]
Hi @NNoureen - continuous integration was failing because `gridGraphics` was not on the project Docker image. I made commits to your branch (https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/506/commits/17af98bb3e31cf9da3259aaba712d0406376d32a and https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/506/commits/9884ca3fe0aa176cbd5885155c97638180091ca2) to get that package installed, so it is a good idea to pull the changes I've made locally before making any further modifications if required. Thank you!

**Comment by NNoureen** [comment]
Thanks. We have pulled the changes.

**Comment by NNoureen** [comment]
Do we need some other changes before merging this branch?

**Comment by NNoureen** [comment]
Sure. Thanks

**Comment by jaclyn-taroni** [comment]
I've updated the branch - I will merge after CI finishes. Thanks all!

**Comment by NNoureen** [comment]
Thanks

**Comment by NNoureen** [comment]
Sure. Thanks. Now we will do the other pull request for the 2nd comparison script. 
Thanks to all the reviewers.

**Pull request merged**